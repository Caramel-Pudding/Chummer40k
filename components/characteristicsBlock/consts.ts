export enum BCCharacteristics {
  WeaponSkill = "Weapon Skill",
  BallisticSkill = "Ballistic Skill",
  Strength = "Strength",
  Toughness = "Toughness",
  Agility = "Agility",
  Intelligence = "Intelligence",
  Perception = "Perception",
  Willpower = "Willpower",
  Fellowship = "Fellowship",
  Infamy = "Infamy",
}
