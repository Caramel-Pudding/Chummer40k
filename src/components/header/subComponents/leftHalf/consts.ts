import { Route } from "./types";

export const routes: Route[] = [
  { url: "/", name: "Main" },
  { url: "/inventory", name: "Inventory" },
];
