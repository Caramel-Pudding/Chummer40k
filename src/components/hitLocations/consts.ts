import { TopPosition, LeftPosition } from "./types";

export const xPositions: Record<string, LeftPosition> = {
  left: "left-6",
  center: "left-20",
  right: "left-36",
};

export const yPositions: Record<string, TopPosition> = {
  top: "top-8",
  center: "top-24",
  bottom: "top-52",
};
