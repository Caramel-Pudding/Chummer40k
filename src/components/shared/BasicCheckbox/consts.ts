export enum BasicCheckboxTestIds {
  checkbox = "BASIC-CHECKBOX-TEST-ID",
}
