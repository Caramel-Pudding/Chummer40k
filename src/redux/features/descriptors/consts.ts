export enum Race {
  Human = "Human",
  ChaosSpaceMarine = "Chaos Space Marine",
}

export enum HumanArchetype {
  Apostate = "Apostate",
  Heretek = "Heretek",
  Renegade = "Renegade",
  Psyker = "Psyker",
  QSalMagisterImmaterial = "Q'Sal Magister Immaterial",
  IdolitrexMagosofForgePolix = "Idolitrex Magos of Forge Polix",
  XuruntFrostFather = "Xurunt Frost Father",
  ChemHunterOfMessia = "Chem-Hunter of Messia",
  PiratePrinceoftheRaggedHelix = "Pirate Prince of the Ragged Helix",
  FleshShaperOfMelancholia = "Flesh Shaper of Melancholia",
  WrithingWorldSorcererKing = "Writhing World Sorcerer-King",
  DeathPriestOfMire = "Death Priest of Mire",
  Plaguemeister = "Plaguemeister",
}

export enum CSMArchetype {
  Champion = "Champion",
  Chosen = "Chosen",
  Forsaken = "Forsaken",
  Sorcerer = "Sorcerer",
  ThousandSonsSorcerer = "Thousand Sons Sorcerer",
  AlphaLegionMarine = "Alpha Legion Marine",
  KhorneBerserker = "Khorne Berserker",
  NightLordsChaosMarine = "Night Lords Chaos Marine",
  NoiseMarine = "Noise Marine",
  DarkApostle = "Dark Apostle",
  PlagueMarine = "Plague Marine",
  Warpsmith = "Warpsmith",
  VeteranOfTheLongWar = "Veteran of the Long War",
}

export type Archetype = HumanArchetype | CSMArchetype;

export enum Pride {
  Beauty = "Beauty",
  Charm = "Charm",
  Craftsmanship = "Craftsmanship",
  Devotion = "Devotion",
  Fortitude = "Fortitude",
  Foresight = "Foresight",
  Logic = "Logic",
  MartialProwess = "Martial Prowess",
  Grace = "Grace",
  Wealth = "Wealth",
}

export enum Disgrace {
  Betrayal = "Betrayal",
  Deceit = "Deceit",
  Dread = "Dread",
  Destruction = "Destruction",
  Gluttony = "Gluttony",
  Greed = "Greed",
  Hubris = "Hubris",
  Regret = "Regret",
  Waste = "Waste",
  Wrath = "Wrath",
}

export enum Motivation {
  Arcane = "Arcane",
  Ascendancy = "Ascendancy",
  Dominion = "Dominion",
  Immortality = "Immortality",
  Innovation = "Innovation",
  Legacy = "Legacy",
  Nihilism = "Nihilism",
  Perfection = "Perfection",
  Vengeance = "Vengeance",
  Violence = "Violence",
}

export enum Alignment {
  Unaligned = "Unaligned",
  Slaanesh = "Slaanesh",
  Nurgle = "Nurgle",
  Khorne = "Khorne",
  Tzeentch = "Tzeentch",
  Outcast = "???",
}

export type DescriptorsUnion =
  | Race
  | Archetype
  | Pride
  | Disgrace
  | Motivation
  | Alignment;
