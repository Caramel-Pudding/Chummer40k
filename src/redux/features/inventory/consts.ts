export enum ItemType {
  Weapons = "weapons",
  Armor = "armor",
  Items = "items",
}
