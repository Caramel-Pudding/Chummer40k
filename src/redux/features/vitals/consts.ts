export enum HitLocation {
  Head = "Head",
  LeftArm = "Left Arm",
  Torso = "Torso",
  RightArm = "Right Arm",
  LeftLeg = "Left Leg",
  RightLeg = "Right Leg",
}
