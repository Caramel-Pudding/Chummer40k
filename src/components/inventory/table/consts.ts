import { ItemType } from "@/redux/features/inventory/consts";

export const tableHeaders = {
  [ItemType.Weapons]: [
    "Name",
    "Class",
    "Range",
    "RoF",
    "Dam.",
    "Pen.",
    "Clip",
    "Rld.",
    "Spc.",
    "Wt.",
    "Av.",
    "Craf.",
  ],
  [ItemType.Armor]: ["Name", "Loc.", "AP", "Spc.", "Wt.", "Av.", "Craf."],
  [ItemType.Other]: ["Name", "Spc.", "Wt.", "Av.", "Craf."],
};
