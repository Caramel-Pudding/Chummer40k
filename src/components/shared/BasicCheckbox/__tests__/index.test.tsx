import React, { FC, useState } from "react";
import { render, cleanup } from "@/tests/test-utils";

import { BasicCheckbox, BasicCheckboxProps } from "..";
import { BasicCheckboxTestIds } from "../consts";

describe("<BasicCheckbox/>", () => {
  afterEach(cleanup);

  const testLabel = "TEST";
  const TestWrapper: FC<Partial<BasicCheckboxProps>> = ({
    checked = false,
  }) => {
    const [isChecked, setIsChecked] = useState(checked);

    return (
      <BasicCheckbox
        checked={isChecked}
        handler={setIsChecked}
        labelText={testLabel}
      />
    );
  };

  it("Basic snapshot test", () => {
    // * #TEST: ARRANGE
    const { container } = render(<TestWrapper />, {});
    // * #TEST: ASSERT
    expect(container).toMatchSnapshot();
  });

  it("should render proper label", () => {
    // * #TEST: ARRANGE
    const { container, getByText } = render(<TestWrapper />, {});
    const label = getByText(testLabel);
    // * #TEST: ASSERT
    expect(label).toHaveTextContent(testLabel);
    expect(container).toMatchSnapshot();
  });

  it("should render undefined label as ''", () => {
    // * #TEST: ARRANGE
    const strange = render(<TestWrapper labelText={undefined} />, {});
    const common = render(<TestWrapper labelText="" />, {});

    // * #TEST: ASSERT
    expect(strange.container).toEqual(common.container);
    expect(strange.container).toMatchSnapshot();
    expect(common.container).toMatchSnapshot();
  });

  it("should turn on by click", () => {
    // * #TEST: ARRANGE
    const { container, getByTestId } = render(
      <TestWrapper checked={false} />,
      {}
    );
    const checkbox: HTMLInputElement = getByTestId(
      BasicCheckboxTestIds.checkbox
    ) as HTMLInputElement;

    expect(checkbox.checked).toEqual(false);
    expect(container).toMatchSnapshot();

    // * #TEST: ACT
    checkbox.click();

    // * #TEST: ASSERT
    expect(checkbox.checked).toEqual(true);
    expect(container).toMatchSnapshot();
  });

  it("should turn off by click", () => {
    // * #TEST: ARRANGE
    const { container, getByTestId } = render(<TestWrapper checked />, {});
    const checkbox: HTMLInputElement = getByTestId(
      BasicCheckboxTestIds.checkbox
    ) as HTMLInputElement;

    expect(checkbox.checked).toEqual(true);
    expect(container).toMatchSnapshot();

    // * #TEST: ACT
    checkbox.click();

    // * #TEST: ASSERT
    expect(checkbox.checked).toEqual(false);
    expect(container).toMatchSnapshot();
  });
});
