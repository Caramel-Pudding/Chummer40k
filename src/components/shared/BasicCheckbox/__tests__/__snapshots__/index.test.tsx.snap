// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicCheckbox/> Basic snapshot test 1`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should render proper label 1`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should render undefined label as '' 1`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should render undefined label as '' 2`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should turn off by click 1`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      checked=""
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should turn off by click 2`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      checked=""
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should turn on by click 1`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<BasicCheckbox/> should turn on by click 2`] = `
<div>
  <label
    class=""
  >
    TEST
    <input
      class="ml-2"
      data-testid="BASIC-CHECKBOX-TEST-ID"
      type="checkbox"
    />
  </label>
</div>
`;
